### ISSUE TITLE
Core should handle message splitting

### ISSUE THREAD

**Issue by subliun** [irungentoo/toxcore#1407]

**Comment by suhr** [comment]
Should it?

**Comment by Impyy** [comment]
No, core shouldn't try to interpret the content of the data that's being sent as messages in any way.
I think this has been discussed before, all core should take care of is sending the data.

**Comment by subliun** [comment]
It doesn't have to interpret the content of the data that's being sent. Core currently doesn't take care of sending the data and makes client authors deal with limitations that they shouldn't have to care about.

**Comment by suhr** [comment]
Yes it does have to. You can't properly split data without knowing what's inside.

**Comment by subliun** [comment]
@suhr see edit

**Comment by suhr** [comment]
So, you want split+merge in toxcore?

**Comment by subliun** [comment]
Yes. Perhaps I could have titled the issue better.

**Comment by o-jasper** [comment]
Sure it can properly split the data without knowing what is inside? All it needs to know is some bunch of data should be called with a callback later, whole, and to then send over the data, and only once entire thing is over, do the callback on the receiving side. Of course, there should be a way to enforce a max size.(so that it is not easy to use as memory-hogging attack)

My opinion in how to do callbacks like that #1403, of course.

**Comment by suhr** [comment]
> callback

Carefully, callbacks may cause headache with the ownership model (C++/Rust).

**Comment by GrayHatter** [comment]
I'm actually in full agreement with this one. This issue needs some more attention.
